### PR TITLE
Retire bom-2.222.x

### LINKED ISSUE #586: Retire 2.222.x BOM line
`README.md` states:

> The UC currently maintains releases for the past 400 days so it is reasonable to retire BOMs for lines older than that.

2.222.4 was released on May 24, 2020, which was 430 days ago. Therefore it is reasonable to retire the 2.222.x BOM line.

### PR DESCRIPTION
**Pull request by jetersen**
fixes #586 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Comment by jetersen** [comment]
right forgot the sample ... I was searching for bom-2.222 😅

**Comment by timja** [comment]
weird, 

```

[2021-07-29T10:27:19.220Z] [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.679 s <<< FAILURE! - in io.jenkins.tools.bom.sample.DeclarativePipelineTest

[2021-07-29T10:27:19.220Z] [ERROR] io.jenkins.tools.bom.sample.DeclarativePipelineTest.smokes  Time elapsed: 4.675 s  <<< FAILURE!

[2021-07-29T10:27:19.220Z] java.lang.AssertionError: 

[2021-07-29T10:27:19.220Z] unexpected build status; build log was:

[2021-07-29T10:27:19.220Z] ------

[2021-07-29T10:27:19.220Z] Started

[2021-07-29T10:27:19.220Z] Running in Durability level: MAX_SURVIVABILITY

[2021-07-29T10:27:19.220Z] [Pipeline] Start of Pipeline

[2021-07-29T10:27:19.220Z] [Pipeline] End of Pipeline

[2021-07-29T10:27:19.220Z] java.lang.NullPointerException

[2021-07-29T10:27:19.220Z] 	at org.jenkinsci.plugins.pipeline.modeldefinition.agent.DeclarativeAgentDescriptor.instanceForDescriptor(DeclarativeAgentDescriptor.java:153)

[2021-07-29T10:27:19.220Z] 	at org.jenkinsci.plugins.pipeline.modeldefinition.agent.DeclarativeAgentDescriptor$instanceForDescriptor$2.call(Unknown Source)

[2021-07-29T10:27:19.220Z] 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)

[2021-07-29T10:27:19.220Z] 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)

[2021-07-29T10:27:19.220Z] 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133)

[2021-07-29T10:27:19.220Z] 	at org.jenkinsci.plugins.pipeline.modeldefinition.model.Agent.getDeclarativeAgent(Agent.groovy:93)

[2021-07-29T10:27:19.220Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

[2021-07-29T10:27:19.220Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

[2021-07-29T10:27:19.220Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

[2021-07-29T10:27:19.220Z] 	at java.lang.reflect.Method.invoke(Method.java:498)

[2021-07-29T10:27:19.220Z] 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)

[2021-07-29T10:27:19.220Z] 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)

[2021-07-29T10:27:19.220Z] 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)

[2021-07-29T10:27:19.220Z] 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
```

**Comment by timja** [comment]
I can reproduce in IDE, https://github.com/jenkinsci/bom/blob/master/sample-plugin/src/test/java/io/jenkins/tools/bom/sample/DeclarativePipelineTest.java

Fails on 2.277.4, but not on 2.289.1

File: bom-latest/pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
         <checks-api.version>1.7.0</checks-api.version>
         <configuration-as-code-plugin.version>1.51</configuration-as-code-plugin.version>
         <git-plugin.version>4.7.2</git-plugin.version>
-        <pipeline-model-definition-plugin.version>1.8.5</pipeline-model-definition-plugin.version>
+        <pipeline-model-definition-plugin.version>1.9.0</pipeline-model-definition-plugin.version>
```
Comment (jglick) [inline bom-latest/pom.xml:15]: subsumes #585

**Review by jglick — APPROVED** [review]
Looks OK, if we can straighten out the CI failures.

**Comment by basil** [comment]
It doesn't look like `pipeline-model-definition` 1.9.0 works at all on 2.277. I filed [JENKINS-66247](https://issues.jenkins.io/browse/JENKINS-66247) to track this. In the meantime, let's upgrade `pipeline-model-definition` to 1.9.0 only on the 2.289.x and later BOM lines pending the resolution of JENKINS-66247.

**Comment by jetersen** [comment]
@basil thanks for carrying it over the finish line! 👏

**Comment by bitwiseman** [comment]
@basil  Strange, looks like https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/442 passed.  So, it must work in some form... Still, I approve this method of moving forward.

**Comment by jglick** [comment]
Tracked down to https://github.com/jenkinsci/lib-annotation-indexer/pull/10.

**Comment by bitwiseman** [comment]
Thanks Jesse! 
@basil @jetersen I'm currently releasing v1.9.1 that should work on all versions if you're willing to wait.

**Comment by timja** [comment]
Test flaked on a 180 second timeout, passed on the retry:
```
 org.jenkinsci.plugins.workflow.CpsDefaultGroovyMethodsTest
```

**Pull request merged**